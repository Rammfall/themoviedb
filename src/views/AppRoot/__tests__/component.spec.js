import React from 'react'
import { shallow } from 'enzyme'

import AppRootComponent from '../component'

describe('AppRootComponent', () => {
  const defaultProps = {
    isLoggedIn: false
  }

  describe('with default props', () => {
    it('matches snapshot', () => {
      // eslint-disable-next-line react/jsx-props-no-spreading
      const component = shallow(<AppRootComponent {...defaultProps} />)

      expect(component).toMatchSnapshot()
    })
  })

  describe('when isLoggedIn true', () => {
    it('matches snapshot', () => {
      const component = shallow(<AppRootComponent isLoggedIn />)

      expect(component).toMatchSnapshot()
    })
  })
})
