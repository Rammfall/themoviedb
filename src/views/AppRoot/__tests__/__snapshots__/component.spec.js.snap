// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRootComponent when isLoggedIn true matches snapshot 1`] = `
<Component>
  <Connect(Route)
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(LoginPage)",
        "type": [Function],
      }
    }
    exact={true}
    path="/login"
  />
  <Connect(Route)
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(withRouter(DashboardPage))",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
    privateRoute={true}
  />
  <Connect(Route)
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(withRouter(ListsPage))",
        "type": [Function],
      }
    }
    exact={true}
    path="/lists"
    privateRoute={true}
  />
</Component>
`;

exports[`AppRootComponent with default props matches snapshot 1`] = `
<Component>
  <Connect(Route)
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(LoginPage)",
        "type": [Function],
      }
    }
    exact={true}
    path="/login"
  />
  <Connect(Route)
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(withRouter(DashboardPage))",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
    privateRoute={true}
  />
  <Connect(Route)
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(withRouter(ListsPage))",
        "type": [Function],
      }
    }
    exact={true}
    path="/lists"
    privateRoute={true}
  />
</Component>
`;
